### PR TITLE
Break out the Curating the Editor Experience doc into its own How-to Guides section

### PR DESCRIPTION
**Pull request by ndiego**
This PR is an initial implementation of the concept outlined in https://github.com/WordPress/gutenberg/issues/56170. 

> V1: Divide and solidify
>
> Break each logical section into individual pages (locking APIs, providing default controls/options, etc) and add in more content/links to articles/video that we already have from the Developer Blog (no new content!).

I have taken the [Curating the Editor Experience](https://developer.wordpress.org/block-editor/how-to-guides/curating-the-editor-experience/) doc and made it its own section in How-to Guides. I have then broken the doc up into five additional docs. 

1. Block locking
1. Patterns
1. theme.json
1. Filters and hooks
1. Disabling Editor functionality

The last section is a bit of a catch-all that covers miscellaneous curation techniques not covered in the other sections. I did add some new content here about disabling block variations and styles.

Let me know what you think of this initial iteration. The next step will be to go through each sub-doc and overhaul the content with additional information, new examples, etc.

### CODE REVIEW

**Review by juanmaguitar — APPROVED** [review]

File: docs/how-to-guides/curating-the-editor-experience/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Curating the Editor Experience
+
+Curating the editing experience in WordPress is important because it allows you to streamline the editing process, ensuring consistency and alignment with the site's style and branding guidelines. It also makes it easier for users to create and manage content effectively without accidental modifications or layout changes. This leads to a more efficient, user-friendly, and visually coherent website.
```
Comment (annezazu) [inline docs/how-to-guides/curating-the-editor-experience/README.md]: >This leads to a more efficient, user-friendly, and visually coherent website.

I might say this leads to a more efficient and personalized experience. Right now, this reads like we're saying it makes a more efficient, user friendly website :)

File: docs/how-to-guides/curating-the-editor-experience/theme-json.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
+# theme.json
```
Comment (annezazu) [inline docs/how-to-guides/curating-the-editor-experience/theme-json.md:1]: Find the lower case t to throw me off here. Any reason not to capitalize?

**Review by annezazu — APPROVED** [review]
Approving but left two minor comments :) Thank you for moving this forward.

File: docs/how-to-guides/curating-the-editor-experience/theme-json.md (hunk 2 above)
Comment (justintadlock) [reply docs/how-to-guides/curating-the-editor-experience/theme-json.md:1]: I haven't tested across different systems, but I'd always err on the side of preserving the correct case for filenames.

File: docs/how-to-guides/curating-the-editor-experience/README.md (hunk 1 above)
Comment (ndiego) [reply docs/how-to-guides/curating-the-editor-experience/README.md]: Good catch, just fixed.

File: docs/how-to-guides/curating-the-editor-experience/theme-json.md (hunk 2 above)
Comment (ndiego) [reply docs/how-to-guides/curating-the-editor-experience/theme-json.md:1]: > I haven't tested across different systems, but I'd always err on the side of preserving the correct case for filenames.

Yeah, we are doing the same thing with other common filenames like block.json, functions.php, etc. That said, we do need to comb through the Handbook and standardize this.

**Comment by annezazu** [comment]
🥳 Thanks for getting this done!

**Pull request merged**